full/full_coupler_mod.F90

```diff
@@ -0,0 +1,1586 @@
+!
+
+  use flux_exchange_mod,       only: flux_exchange_init, gas_exchange_init, sfc_boundary_layer
+  use flux_exchange_mod,       only: generate_sfc_xgrid, send_ice_mask_sic
+  use flux_exchange_mod,       only: flux_down_from_atmos, flux_up_to_atmos
+  use flux_exchange_mod,       only: flux_land_to_ice, flux_ice_to_ocean, flux_ocean_to_ice
+  use flux_exchange_mod,       only: flux_ice_to_ocean_finish, flux_ocean_to_ice_finish
+  use flux_exchange_mod,       only: flux_check_stocks, flux_init_stocks
+  use flux_exchange_mod,       only: flux_ocean_from_ice_stocks, flux_ice_to_ocean_stocks
+  use flux_exchange_mod,       only: flux_atmos_to_ocean, flux_ex_arrays_dealloc
+
+  use atmos_tracer_driver_mod, only: atmos_tracer_driver_gather_data
+
+  use iso_fortran_env
+
+  implicit none
+  private
+  
+  public :: fmsconstants_init
+  public :: update_atmos_model_dynamics, update_atmos_model_down, update_atmos_model_up
+  public :: update_atmos_model_radiation, update_atmos_model_state
+  public :: update_land_model_fast, update_land_model_slow
+  public :: update_ice_model_fast, set_ice_surface_fields
+  public :: unpack_ocean_ice_boundary, exchange_slow_to_fast_ice
+  public :: ice_model_fast_cleanup, unpack_land_ice_boundary
+  public :: exchange_fast_to_slow_ice, update_ice_model_slow
+  public :: update_ocean_model, update_slow_ice_and_ocean
+  public :: sfc_boundary_layer, generate_sfc_xgrid, send_ice_mask_sic
+  public :: flux_down_from_atmos, flux_up_to_atmos
+  public :: flux_land_to_ice, flux_ice_to_ocean, flux_ocean_to_ice
+  public :: flux_ice_to_ocean_finish, flux_ocean_to_ice_finish
+  public :: flux_check_stocks, flux_init_stocks
+  public :: flux_ocean_from_ice_stocks, flux_ice_to_ocean_stocks
+  public :: flux_atmos_to_ocean, flux_ex_arrays_dealloc
+  public :: atmos_tracer_driver_gather_data
+
+  public :: atmos_model_restart, land_model_restart, ice_model_restart, ocean_model_restart
+
+  public :: atmos_data_type_chksum,   lnd_ice_atm_bnd_type_chksum
+  public :: lnd_atm_bnd_type_chksum,  ice_atm_bnd_type_chksum
+  public :: atm_lnd_bnd_type_chksum,  land_data_type_chksum
+  public :: ice_data_type_chksum,     ocn_ice_bnd_type_chksum
+  public :: atm_ice_bnd_type_chksum,  lnd_ice_bnd_type_chksum
+  public :: ocean_ice_boundary_type,  atmos_ice_boundary_type
+  public :: ocean_public_type_chksum, ice_ocn_bnd_type_chksum
+
+  public :: coupler_init, coupler_end, coupler_restart
+  public :: coupler_chksum, atmos_ice_land_chksum, slow_ice_chksum, ocean_chksum, full_coupler_set_clock_ids
+
+!-----------------------------------------------------------------------
+
+#ifdef FULL_COUPLER_VERSION_
+  character(len=*), parameter, public :: version = FULL_COUPLER_VERSION_
+#else
+  character(len=*), parameter, public :: version = 'FULL_COUPLER_MOD'
+#endif
+
+#ifdef FULL_COUPLER_TAG_
+  character(len=*), parameter, public :: tag = FULL_COUPLER_TAG_
+#else
+  character(len=*), parameter, public :: tag = 'unknown'
+#endif
```

Comment:
We no longer use tag in any meaningful way and you can delete these lines of code and any use of it later in the file.